holoviews/element/util.py

```diff
@@ -30,3 +44,98 @@ def compute_edges(edges):
         raise ValueError('Centered bins have to be of equal width.')
     edges -= width/2.
     return np.concatenate([edges, [edges[-1]+width]])
+
+
+def reduce_fn(x):
+    """
+    Aggregation function to get the first non-zero value.
+    """
+    values = x.values if pd and isinstance(x, pd.Series) else x
+    for v in values:
+        if not is_nan(v):
+            return v
+    return np.NaN
+
+
+class categorical_aggregate2d(ElementOperation):
```

Comment:
Looks great! I was just wondering if you want to keep this class in ``util`` or move it to ``operations.element``?